src/Http/WebUtilities/src/HttpRequestStreamReader.cs

```diff
@@ -370,5 +370,19 @@ private async Task<int> ReadIntoBufferAsync()
 
             return _charsRead;
         }
+
+        public async override Task<string> ReadToEndAsync()
+        {
+            StringBuilder sb = new StringBuilder(_charsRead - _charBufferIndex);
```

Comment:
Wouldn't this make it so the string builder is guaranteed to resize if the first call to ReadIntoBufferAsync returns a size > 0?

Reply:
also nit: we prefer using var instead of types:
```suggestion
            var sb = new StringBuilder(_charsRead - _charBufferIndex);
```

Reply:
Hi @jkotalik,
I think you're right. As I mentioned in the PR description, this comes from the `StreamReader.ReadToEndAsync` implementation, which I assumed to be correct, but maybe it could be improved.